src/wazuh_modules/syscollector/src/syscollectorImp.cpp

```diff
@@ -1630,37 +1636,71 @@ void Syscollector::syncLoop(std::unique_lock<std::mutex>& lock)
         sync();
     }
 
-    while (!m_cv.wait_for(lock, std::chrono::seconds{m_intervalValue}, [&]()
+    while (!m_cv.wait_for(lock, std::chrono::seconds{m_adjustedIntervalValue ? m_adjustedIntervalValue : m_intervalValue}, [&]()
 {
-    return m_stopping;
-}))
+    return (m_stopping);
+    }))
     {
-        scan();
-        sync();
+        if (!m_lastScanOnDemand)
+        {
+            scan();
+            sync();
+            m_adjustedIntervalValue = 0;
+        }
+        else
+        {
+            m_logFunction(SYS_LOG_DEBUG_VERBOSE, "Syscollector scan interval adjusted due to a recent on demand scan.");
+            std::time_t current_time = std::time(nullptr);
+            m_adjustedIntervalValue = m_intervalValue - (current_time - m_lastScanOnDemandTime);
+        }
+
+        m_lastScanOnDemand = false;
     }
     m_spRsync.reset(nullptr);
     m_spDBSync.reset(nullptr);
 }
 
-void Syscollector::push(const std::string& data)
+void Syscollector::runOnDemandScan()
 {
     std::unique_lock<std::mutex> lock{m_mutex};
 
     if (!m_stopping)
     {
-        auto rawData{data};
-        Utils::replaceFirst(rawData, "dbsync ", "");
-        const auto buff{reinterpret_cast<const uint8_t*>(rawData.c_str())};
+        m_logFunction(SYS_LOG_DEBUG, "Syscollector scan on demand requested.");
+        scan();
+        sync();
+        m_lastScanOnDemand = true;
+        m_lastScanOnDemandTime = std::time(nullptr);
+    }
+}
 
-        try
+void Syscollector::push(const std::string& data)
+{
+    std::unique_lock<std::mutex> lock{m_mutex};
+
+    if (!m_stopping)
+    {
+        if (0 == data.compare("syscollector_run"))
```

Comment:
Change the message to perform the on-demand scan when we receive `syscollector_scan start`.

Reply:
Message updated !